source/wifi/wifi_hal.c

```diff
@@ -8719,9 +8720,174 @@ INT wifi_createVAP(wifi_radio_index_t index, wifi_vap_info_map_t *map)
     return RETURN_OK;
 }
 
+int parse_channel_list_int_arr(int radioIndex, char *pchannels, wifi_channels_list_t* chlistptr)
+{
+
+    char *token;
+    const char s[2] = ",";
+    int i =0, count =0;
+
+    /* get the first token */
+    token = strtok(pchannels, s);
+
+    /* walk through other tokens */
+    while( token != NULL ) {
+       count++;
+       chlistptr->channels_list[i] = atoi(token);
+       token = strtok(NULL, s);
+   }
+   return count;
+}
+
+static int getRadioCapabilities(int radioIndex, wifi_radio_capabilities_t *rcap)
+{
+
+    INT status;
+    wifi_channels_list_t *chlistp;
+    unsigned int *uarray;
+    CHAR output_string[64];
+    CHAR pchannels[128];
+
+    if(rcap == NULL)
+    {
+        return RETURN_ERR;
+    }
+
+    rcap->numSupportedFreqBand = 1;
+    if (1 == radioIndex)
+      rcap->band[0] = WIFI_FREQUENCY_5_BAND;
+    else
+      rcap->band[0] = WIFI_FREQUENCY_2_4_BAND;
+
+
+    chlistp = &(rcap->channel_list[0]);
+    memset(pchannels, 0, sizeof(pchannels));
+
+    /* possible number of radio channels */
+    status = wifi_getRadioPossibleChannels(radioIndex, pchannels);
+    {
+         printf("[wifi_hal dbg] : func[%s] line[%d] error_ret[%d] radio_index[%d] output[%s]\n", __FUNCTION__, __LINE__, status, radioIndex, output_string);
+    }
+    /* Number of channels and list*/
+    chlistp->num_channels = parse_channel_list_int_arr(radioIndex, pchannels, chlistp);
+
+    /* autoChannelSupported */
+    /* always ON with wifi_getRadioAutoChannelSupported */
+    rcap->autoChannelSupported = TRUE;
+
+    /* DCSSupported */
+    /* always ON with wifi_getRadioDCSSupported */
+    rcap->DCSSupported = TRUE;
+
+    /* zeroDFSSupported - TBD */
+    rcap->zeroDFSSupported = FALSE;
+
+    /* Supported Country List*/
+    memset(output_string, 0, sizeof(output_string));
+    status = wifi_getRadioCountryCode(radioIndex, output_string);
+    if( status != 0 ) {
+        printf("[wifi_hal dbg] : func[%s] line[%d] error_ret[%d] radio_index[%d] output[%s]\n", __FUNCTION__, __LINE__, status, radioIndex, output_string);
+        return RETURN_ERR;
+    } else {
+        printf("[wifi_hal dbg] : func[%s] line[%d], output [%s]\n", __FUNCTION__, __LINE__, output_string);
+    }
+    output_string[strlen(output_string) - 1] = '\0';
+    if(!strcmp(output_string,"US")){
+        rcap->countrySupported[0] = wifi_countrycode_US;
+        rcap->countrySupported[1] = wifi_countrycode_CA;
+    } else if (!strcmp(output_string,"CA")) {
+        rcap->countrySupported[0] = wifi_countrycode_CA;
+        rcap->countrySupported[1] = wifi_countrycode_US;
+    } else {
+        printf("[wifi_hal dbg] : func[%s] line[%d] radio_index[%d] Invalid Country [%s]\n", __FUNCTION__, __LINE__, radioIndex, output_string);
+    }
+
+    rcap->numcountrySupported = 2;
+
+    /* csi */
+    rcap->csi.maxDevices = 8;
+    rcap->csi.soudingFrameSupported = TRUE;
+
+    snprintf(rcap->ifaceName, 64, "wifi%d_0", (radioIndex == 0) ? 2 : 0);
```

Comment:
i think, radio interface name wlan0/wlan1.
with this code, radio interface name might be wifi0_0/wifi2_0.
please clarify

Reply:
modified accordingly